client/me/purchases/payment-methods/existing-credit-card.tsx

```diff
@@ -0,0 +1,400 @@
+import { Gridicon } from '@automattic/components';
+import { Button, FormStatus, useLineItems, useFormStatus } from '@automattic/composite-checkout';
+import { PaymentLogo } from '@automattic/wpcom-checkout/src/payment-method-logos';
+import { SummaryLine, SummaryDetails } from '@automattic/wpcom-checkout/src/summary-details';
+import styled from '@emotion/styled';
+import { sprintf } from '@wordpress/i18n';
+import { useI18n } from '@wordpress/react-i18n';
+import debugFactory from 'debug';
+import { useTranslate } from 'i18n-calypso';
+import { Fragment, useCallback, useState } from 'react';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+import { useDispatch } from 'react-redux';
+import { getPaymentMethodSummary } from 'calypso/lib/checkout/payment-methods';
+import wpcom from 'calypso/lib/wp';
+import { errorNotice } from 'calypso/state/notices/actions';
+import PaymentMethodEditButton from './components/payment-method-edit-button';
+import PaymentMethodEditDialog from './components/payment-method-edit-dialog';
+import RenderEditFormFields from './components/payment-method-edit-form-fields';
+import type { PaymentMethod, ProcessPayment, LineItem } from '@automattic/composite-checkout';
+
+const debug = debugFactory( 'calypso:existing-card-payment-method' );
+
+// Disabling this to make migration easier
+/* eslint-disable @typescript-eslint/no-use-before-define */
+
+async function fetchTaxInfo(
+	storedDetailsId: string
+): Promise< { tax_postal_code: string; tax_country_code: string; is_tax_info_set: boolean } > {
+	return await wpcom.req.get( `/me/payment-methods/${ storedDetailsId }/tax-location` );
+}
+
+async function setTaxInfo(
+	storedDetailsId: string,
+	taxPostalCode: string,
+	taxCountryCode: string
+): Promise< { tax_postal_code: string; tax_country_code: string } > {
+	return await wpcom.req.post( {
+		path: `/me/payment-methods/${ storedDetailsId }/tax-location`,
+		body: {
+			tax_country_code: taxCountryCode,
+			tax_postal_code: taxPostalCode,
+		},
+	} );
+}
+
+function joinNonEmptyValues( joinString: string, ...values: any ) {
```

Comment:
It's not a great idea to use the `any` type since it turns off type checking for that value. In this case I think you can safely say what types of data `values` should be

```suggestion
function joinNonEmptyValues( joinString: string, ...values: string[] ) {
```

Technically the way this function is written, it also supports undefined values, but if you look at how it's being used below, `postalCodeValue` and `countryCodeValue` will always be strings.

Reply:
I've corrected the type on this.